pkg/cli/cli.go

```diff
@@ -19,41 +20,29 @@ type cliModule struct {
 	Module
 }
 
-type kernelSettings struct {
-	KillTimeout time.Duration `cfg:"killTimeout" default:"10s"`
-}
-
 func newCliModule(module Module) *cliModule {
 	return &cliModule{
 		Module: module,
 	}
 }
 
 func Run(module Module) {
-	configOptions := []cfg.Option{
-		cfg.WithErrorHandlers(defaultErrorHandler),
-		cfg.WithConfigFile("./config.dist.yml", "yml"),
-		cfg.WithConfigFileFlag("config"),
-	}
-
-	config := cfg.New()
-	if err := config.Option(configOptions...); err != nil {
-		defaultErrorHandler(err, "can not initialize the config")
-	}
-
-	logger, err := newCliLogger()
-	if err != nil {
-		defaultErrorHandler(err, "can not initialize the logger")
-	}
-
-	if err := module.Boot(config, logger); err != nil {
-		defaultErrorHandler(err, "can not boot the module")
-	}
-
-	settings := &kernelSettings{}
-	config.UnmarshalKey("kernel", settings)
-
-	k := kernel.New(config, logger, kernel.KillTimeout(settings.KillTimeout))
+	k := application.New(
```

Comment:
k is now an app and not a kernel, you could rename the variable. But this is only minor nitpicking.

Another thing would be to have more shared initialization logic - the app options you are listing are a subset of the defaults, so maybe we should move them all to the application package and then add the options for a service app to that list - otherwise one might forget to add an option for a CLI app when adding it to a server which should've been added to a CLI app.

A last thing, do we still need the cliModule type? The only use I see is having the module be an essential module, but you can also pass that as an option to Add().

Reply:
Makes sense, applied all of that, but not the last part "remove cliModule".

In fact we don't need it, it is just a handy helper like we have for the `RunConsumer`